### PR TITLE
Remove cxx aliases

### PR DESCRIPTION
**Pull request by hjmjohnson**
## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

File: Utilities/ITKv5Preparation/ReplaceOutdatedMacroNames.sh [hunk 1]
```diff
@@ -30,7 +30,18 @@ echo "WARNING:  This script is not intended to be bullet-proof."
 echo "WARNING:  Please carefully review all changes made to ensure proper behavior."
 }
 
-ReplaceCXXString ITK_NOEXCEPT_OR_THROW ITK_NOEXCEPT
-ReplaceCXXString ITK_HAS_CXX11_STATIC_ASSERT ITK_COMPILER_CXX_STATIC_ASSERT
-ReplaceCXXString ITK_DELETE_FUNCTION ITK_DELETED_FUNCTION
-ReplaceCXXString ITK_HAS_CPP11_ALIGNAS ITK_COMPILER_CXX_ALIGNAS
+ReplaceCXXString ITK_NOEXCEPT_OR_THROW             ITK_NOEXCEPT
```
Comment (jhlegarreta) [inline Utilities/ITKv5Preparation/ReplaceOutdatedMacroNames.sh:33]: Should these contents be in a folder named `Utilities/ITKv6Preparation/ReplaceOutdatedMacroNames.sh` (note the **v6**)?

File: Modules/Core/Common/include/itkMacro.h [hunk 2]
```diff
@@ -1416,31 +1381,80 @@ itkDynamicCastInDebugMode(TSource x)
...
-#    define ITK_NODISCARD(message) [[nodiscard]]
-#  endif
+#if __cplusplus >= 202002L
+#  define ITK_NODISCARD(message) [[nodiscard(message)]]
 #else
-#  define ITK_NODISCARD(message)
+#  define ITK_NODISCARD(message) [[nodiscard]]
 #endif
 
+//-*-*-*
```
Comment (dzenanz) [inline Modules/Core/Common/include/itkMacro.h:1390]: Does this comment have some special meaning?

**Review by thewtex — APPROVED** [review]
:1st_place_medal:

**Review by dzenanz — APPROVED** [review]

File: Modules/Core/Common/include/itkMacro.h (hunk 2 above)
Comment (hjmjohnson) [reply Modules/Core/Common/include/itkMacro.h:1390]: It was there before moved with code from above.

**Pull request merged**